addons/website_event/tests/test_event_menus.py

```diff
@@ -67,3 +67,32 @@ def test_menu_management_frontend(self):
         # re-created from backend
         event.introduction_menu = True
         self._assert_website_menus(event, ['Introduction', 'Location', 'Register'], menus_out=['Community'])
+
+    def test_submenu_url_uniqueness(self):
+        """Ensure that the last part of the menus URL (used to retrieve the right view)
+        are unique when creating two events with same name."""
+        event_1, event_2 = self.env["event.event"].create(
+            [
+                {
+                    "name": "TestEvent",
+                    "date_begin": fields.Datetime.to_string(
+                        datetime.today() + timedelta(days=1)
+                    ),
+                    "date_end": fields.Datetime.to_string(
+                        datetime.today() + timedelta(days=15)
+                    ),
+                    # Skip the register and community menus since they already have a unique URL
+                    "website_menu": True,
+                    "register_menu": False,
+                    "community_menu": False,
+                }
+                for _ in range(2)
+            ]
+        )
+
+        for event_1_menu, event_2_menu in zip(
+            event_1.menu_id.child_id, event_2.menu_id.child_id
+        ):
+            self.assertNotEqual(
+                event_1_menu.url.split("/")[-1], event_2_menu.url.split("/")[-1]
```

Comment:
Would maybe go one step further and check that the end of the url matches a view each time